integrations/weaviate/src/haystack_integrations/document_stores/weaviate/auth.py

```diff
@@ -0,0 +1,170 @@
+from weaviate.auth import AuthClientPassword as WeaviateAuthClientPassword
+
+
+@dataclass
+class AuthCredentials(ABC):
+    """
+    Base class for all auth credentials supported by WeaviateDocumentStore.
+    Can be used to deserialize from dict any of the supported auth credentials.
+    """
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Converts the object to a dictionary representation for serialization.
+        """
+        # We assume all fields are Secret instances
+        _fields = {f.name: getattr(self, f.name).to_dict() for f in fields(self)}
+
+        return default_to_dict(
+            self,
+            **_fields,
+        )
+
+    @staticmethod
+    def from_dict(data: Dict[str, Any]) -> "AuthCredentials":
+        """
+        Converts a dictionary representation to an auth credentials object.
+        """
+        if "type" not in data:
+            msg = "Missing 'type' in serialization data"
+            raise DeserializationError(msg)
+        return _AUTH_CLASSES[data["type"]]._from_dict(data)
+
+    @classmethod
+    @abstractmethod
+    def _from_dict(cls, data: Dict[str, Any]):
+        """
+        Internal method to convert a dictionary representation to an auth credentials object.
+        All subclasses must implement this method.
+        """
+
+    @abstractmethod
+    def resolve_value(self):
+        """
+        Resolves all the secrets in the auth credentials object and returns the corresponding Weaviate object.
+        All subclasses must implement this method.
+        """
+
+
+@dataclass
+class AuthApiKey(AuthCredentials):
+    """
+    AuthCredentials for API key authentication.
+    By default it will load `api_key` from the environment variable `WEAVIATE_API_KEY`.
+    """
+
+    api_key: Secret = field(default_factory=lambda: Secret.from_env_var(["WEAVIATE_API_KEY"]))
+
+    @classmethod
+    def _from_dict(cls, data: Dict[str, Any]) -> "AuthApiKey":
+        deserialize_secrets_inplace(data["init_parameters"], ["api_key"])
+        return default_from_dict(cls, data)
+
+    def resolve_value(self) -> WeaviateAuthApiKey:
+        return WeaviateAuthApiKey(api_key=self.api_key.resolve_value())
+
+
+@dataclass
+class AuthBearerToken(AuthCredentials):
+    """
+    AuthCredentials for Bearer token authentication.
+    By default it will load `access_token` from the environment variable `WEAVIATE_ACCESS_TOKEN`,
+    `expires_in` from the environment variable `WEAVIATE_EXPIRES_IN`, and `refresh_token` from the environment variable
+    `WEAVIATE_REFRESH_TOKEN`.
+    `WEAVIATE_EXPIRES_IN` environment variable is optional, if set must be an integer.
+    `WEAVIATE_REFRESH_TOKEN` environment variable is optional.
+    """
+
+    access_token: Secret = field(default_factory=lambda: Secret.from_env_var(["WEAVIATE_ACCESS_TOKEN"]))
+    expires_in: Secret = field(default_factory=lambda: Secret.from_env_var(["WEAVIATE_EXPIRES_IN"], strict=False))
```

Comment:
This doesn't look like a secret. Let's just make it a regular int that defaults to 60.